### PR TITLE
Port DumpSettingsTest to Kotlin

### PR DESCRIPTION
**Pull request by krzema12**
Part of https://github.com/krzema12/snakeyaml-engine-kmp/issues/269.

### CODE REVIEW

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/api/dump/DumpSettingsTest.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            DumpSettings.builder().setIndent(11)
+        }
+        exception2.message shouldBe "Indent must be at in range 1..10"
+    }
+
+    test("Check corner cases for Indicator Indent") {
+        val exception1 = shouldThrow<EmitterException> {
+            DumpSettings.builder().setIndicatorIndent(-1)
+        }
+        exception1.message shouldBe "Indicator indent must be in range 0..9"
```
Comment (krzema12) [inline src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/api/dump/DumpSettingsTest.kt:101]: ShouldTrrowWithMessage

File: src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/api/dump/DumpSettingsTest.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        val settings = DumpSettings.builder()
+            .setCustomProperty(KeyName("key"), "value")
+            .build()
+        settings.getCustomProperty(KeyName("key")) shouldBe "value"
+        settings.getCustomProperty(KeyName("None")).shouldBeNull()
+    }
+
+    test("Use Core schema by default") {
+        val settings = DumpSettings.builder().build()
+        settings.schema::class shouldBe CoreSchema::class
```
Comment (krzema12) [inline src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/api/dump/DumpSettingsTest.kt:126]: ShouldBeOfType
